### ISSUE TITLE
Elixir-compatible exception structs

### ISSUE THREAD

**Issue by jparise** [pinterest/elixir-thrift#239]
Elixir requires exception structs to define two function in compliance with the Exception behaviour: `exception/1` and `message/1`.

We generate Elixir exception structs via `defexception` for Thrift exception types. That macro always provides an `exception/1` implementation, but `message/1` is only generated if the exception is defined with a `:message` field. When it doesn't, the generated code produces warnings like this:

```
warning: undefined behaviour function message/1 (for behaviour Exception)
```

> message/1 - receives the exception struct and must return its message. Most commonly exceptions have a message field which by default is accessed by this function. However, if an exception does not have a message field, this function must be explicitly implemented.

We can accommodate this requirement in one of two ways:

1. We can insert a `:message` key into our exception structs whenever one doesn't already exist and give it a reasonable default value.
2. We can generate our own `message/1` functions for exceptions, conditionally (based on the existence of a `:message` field) or always.

(1) is the easiest thing to do in terms of code changes, but it has the disadvantage of introduce a struct field that doesn't exist in the underlying Thrift IDL.

(2) doesn't suffer from any obvious disadvantages, although it does appear to involve more complicated code changes based on how we generate the quoted module AST (unless I'm missing a more obvious way to conditionialize it).

**Comment by jparise** [comment]
There's also the related question of what to return from a `message/1` implementation when we don't know anything more specific about the Thrift exception we're producing.

**Comment by jparise** [comment]
Closed by #242
